src/include/mpir_handlemem.h

```diff
@@ -128,6 +128,9 @@ static inline void *MPIR_Handle_indirect_init(void *(**indirect)[],
     char *ptr;
     int i;
 
+    MPIR_Assert(indirect_num_blocks > 0);
+    MPIR_Assert(indirect_num_indices > 0);
+
```

Comment:
The only call to `MPIR_Handle_indirect_init` is in the same file at line 265, where it is called with `HANDLE_NUM_BLOCKS`, `HANDLE_NUM_INDICES`, i.e. constants. So I wonder how klocwork misses that. But I guess the asserts doesn't hurt. In the same vein, don't we need to assert `obj_size > 0`?

Reply:
Okay, I see what's going on. In our internal branch we extended the object allocator to take variables for these two arguments. So probably these warnings and assertions are relevant only to our branch. Shall I close this?

Reply:
I am OK with adding asserts if it helps to clean up the tools. Personally, I like to use the simple `assert` here, since it is more standard, and potentially more recognizable by static analyzers. `MPIR_Assert` can be used for the cases that the authors are not 100% sure, i.e. the case that is possible to happen. @raffenet What do you think?

Reply:
We actually modified `MPIR_Assert` to work better with static checkers: https://github.com/pmodels/mpich/blob/32c1ce36025d9b4706e77bd95f6502168ab99236/src/include/mpir_assert.h#L31

IMO it's OK to continue using it. The only difference is at runtime, an `MPIR_Assert` will do a clean abort rather than just crash and hope `mpiexec` can clean up the mess.

Reply:
Interesting. I don't have strong opinions either way. For the cases that we are adding asserts just to increase readability or to please the analyzer, I just find the plane `assert` is more explicit on this intention.

Reply:
Yes, I see your point that being able to identify assertions that are there to silence static checkers would be nice. But I also don't feel too strongly about it, plus I worry that developers may not care to learn the difference in the long run 😄 .

Reply:
Actually `MPIR_Assert` has been working fine with Klocwork (and Coverity, I believe).

Reply:
Oh I missed Ken's comment with a code snipped above.  Nevermind.